packages/vite-plugin-svelte/src/utils/dependencies.ts

```diff
@@ -102,10 +104,27 @@ function parsePkg(dir: string, silent = false): Pkg | void {
 	}
 }
 
-function isSvelte(pkg: Pkg) {
+function getSvelteDependencyType(pkg: Pkg): SvelteDependencyType | undefined {
+	if (isSvelteComponentLib(pkg)) {
+		return 'component-library';
+	} else if (isSvelteLib(pkg)) {
+		return 'js-library';
+	} else {
+		return undefined;
+	}
+}
+
+function isSvelteComponentLib(pkg: Pkg) {
 	return !!pkg.svelte;
 }
 
+function isSvelteLib(pkg: Pkg) {
+	return (
+		Object.keys(pkg.dependencies || {}).includes('svelte') ||
```

Comment:
can be simplified to pkg.dependencies?.svelte

Reply:
That's a lot better